GiftModels/PremiumDetails.cs

```diff
@@ -52,22 +49,66 @@ public string PremiumChart
         ///This is the KFS Account number used to credit the back the premium amount. 
         /// </summary>
         private string _premiumAccount;
+
         public string PremiumAccount
         {
             get
             {
                 if (string.IsNullOrWhiteSpace(_premiumAccount) && string.IsNullOrWhiteSpace(Xcomment))
                     return null;
-                if (string.IsNullOrWhiteSpace(_premiumAccount) && Xcomment.Contains("-"))
-                    return Xcomment.Substring(Xcomment.IndexOf('-') + 1);
+                if (string.IsNullOrWhiteSpace(_premiumAccount))
+                    InitAccounDetails();
+
                 return _premiumAccount;
             }
             set { _premiumAccount = value; }
         }
 
         /// <summary>
+        ///This is the KFS Sub-account number used to credit the back the premium amount. 
+        /// </summary>
+        private string _premiumSubAccount;
+
+        public string PremiumSubAccount
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(_premiumSubAccount) && string.IsNullOrWhiteSpace(Xcomment))
+                    return null;
+                if (string.IsNullOrWhiteSpace(_premiumAccount))
+                    InitAccounDetails();
+
+                return _premiumSubAccount;
+            }
+            set { _premiumSubAccount = value; }
+        }
+
+        /// <summary>
         /// This field is defaulted from the tms_premium table (W3). It is the monetary value associated with this premium. This value is normally the same amount as the value in the tms_premium table. 
         /// </summary>
         public decimal PremiumAmount { get; set; }
+
+        /// <summary>
+        /// Parse out the KFS chart, Account, and Sub-account (if present) from the XComment field. 
+        /// </summary>
+        private void InitAccounDetails()
+        {
+            const string sPattern = @"\[\[(\w)-(\w{5,7})-?(\w{5})?\]\]";
+            var result = System.Text.RegularExpressions.Regex.Match(Xcomment, sPattern);
+
+            if (result.Groups[1].Length > 0 && result.Groups[1].ToString().Length > 0)

```

Comment:
@jpknoll : My intention was to make sure that we could pull non-empty strings for chart and account.  Checking just the Groups[0] doesn't necessarily tell me that, I think.